### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by andersy005**
~~This is still work in progress ...~~

### CODE REVIEW

**Comment by andersy005** [comment]
@martindurant, @aaronspring this is ready for review when you get a moment. You can view the rendered version from https://intake-thredds--19.org.readthedocs.build/en/19/

File: intake_thredds/source.py [hunk 1]
```diff
@@ -11,31 +11,54 @@
...
+        If True, will print a progress bar. Requires `tqdm <https://github.com/tqdm/tqdm>`__
+        to be installed.
+    metadata : dict or None
+        To associate with this source.
+
+    Examples
+    --------
+    >>> import intake
+    >>> cat_url = 'https://psl.noaa.gov/thredds/catalog.xml'
+    >>> paths = ['Datasets', 'ncep.reanalysis.dailyavgs', 'surface', 'air*sig995*194*.nc']
```
Comment (aaronspring) [inline intake_thredds/source.py:37]:
```suggestion
    >>> paths = ['Datasets', 'ncep.reanalysis.dailyavgs', 'surface', 'air.sig995.194*.nc']
```

**Comment by aaronspring** [comment]
can we give this a nicer name instead of name not found?
image

**Comment by aaronspring** [comment]
I would also add section in the tutorial about the different drivers and simplecache. Should I add this?

File: tests/test_source.py [hunk 2]
```diff
@@ -66,7 +66,9 @@ def test_THREDDSMergedSource(THREDDSMergedSource_cat):
 def test_THREDDSMergedSource_long_short(THREDDSMergedSource_cat, THREDDSMergedSource_cat_short):
     ds = THREDDSMergedSource_cat.to_dask()
     ds_short = THREDDSMergedSource_cat_short.to_dask()
-    xr.testing.assert_equal(ds, ds_short)  # TODO: down load data only compare dims, coords, size
+    xr.testing.assert_equal(
+        ds, ds_short
+    )  # downloads data now, faster only compare dims, coords, size
```
Comment (aaronspring) [inline tests/test_source.py:71]: I didnt find assert_identical_coords_and_size
```suggestion
    for c in ds.coords:
        assert (ds[c] == ds_short[c]).all()
        assert ds.size == ds_short.size
```
I didnt find assert_identical_coords_and_size

**Review by aaronspring — APPROVED** [review]
I really nice this new documentation, see #20 for additional section

**Comment by martindurant** [comment]
Great improvements throughout.
The notebook doesn't show that well on github, but maybe that doesn't matter.

**Comment by andersy005** [comment]
> can we give this a nicer name instead of name not found?
> image


I thought I could just do

```python
catalog = intake.open_thredds_cat(cat_url, name='noaa-ersst-catalog')
```

and intake would take care of the rest in https://github.com/NCAR/intake-thredds/blob/3d5e08c3d784f235f8139b8b1228629138da0eea/intake_thredds/cat.py#L8-L11

For some reason, the name argument doesn't get propagated to the base class and `self.name` stays None no matter what. @martindurant, I'm missing anything here?

**Comment by martindurant** [comment]
Yes, you can pass `name=` to `Catalog`, and `intake.open_thredds_cat is intake_thredds.cat.ThreddsCatalog`. 
Is [this line](https://github.com/NCAR/intake-thredds/blob/3d5e08c3d784f235f8139b8b1228629138da0eea/intake_thredds/cat.py#L30) resetting the value?

**Comment by andersy005** [comment]
> Yes, you can pass `name=` to `Catalog`, and `intake.open_thredds_cat is intake_thredds.cat.ThreddsCatalog`.
> Is [this line](https://github.com/NCAR/intake-thredds/blob/3d5e08c3d784f235f8139b8b1228629138da0eea/intake_thredds/cat.py#L30) resetting the value?

👍🏽 Good catch. Thanks! Should be fixed now

**Comment by andersy005** [comment]
Not sure what's causing the `ModuleNotFoundError: No module named 'msgpack'` error on RTD. It was happening before and I just started seeing it a few minutes ago. Will look into this later today. 

```bash
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/intake-thredds/envs/19/lib/python3.7/site-packages/sphinx/config.py", line 327, in eval_config_file
    execfile_(filename, namespace)
  File "/home/docs/checkouts/readthedocs.org/user_builds/intake-thredds/envs/19/lib/python3.7/site-packages/sphinx/util/pycompat.py", line 88, in execfile_
    exec(code, _globals)
  File "/home/docs/checkouts/readthedocs.org/user_builds/intake-thredds/checkouts/19/docs/source/conf.py", line 17, in <module>
    import intake_thredds
  File "/home/docs/checkouts/readthedocs.org/user_builds/intake-thredds/envs/19/lib/python3.7/site-packages/intake_thredds-0.0.post56-py3.7.egg/intake_thredds/__init__.py", line 8, in <module>
    from .source import THREDDSMergedSource  # noqa: F401
  File "/home/docs/checkouts/readthedocs.org/user_builds/intake-thredds/envs/19/lib/python3.7/site-packages/intake_thredds-0.0.post56-py3.7.egg/intake_thredds/source.py", line 3, in <module>
    from intake_xarray.base import DataSourceMixin
  File "/home/docs/checkouts/readthedocs.org/user_builds/intake-thredds/envs/19/lib/python3.7/site-packages/intake_xarray/__init__.py", line 6, in <module>
    from intake.container import register_container
  File "/home/docs/checkouts/readthedocs.org/user_builds/intake-thredds/envs/19/lib/python3.7/site-packages/intake/container/__init__.py", line 8, in <module>
    from .dataframe import RemoteDataFrame
  File "/home/docs/checkouts/readthedocs.org/user_builds/intake-thredds/envs/19/lib/python3.7/site-packages/intake/container/dataframe.py", line 11, in <module>
    from .base import RemoteSource, get_partition
  File "/home/docs/checkouts/readthedocs.org/user_builds/intake-thredds/envs/19/lib/python3.7/site-packages/intake/container/base.py", line 8, in <module>
    import msgpack
ModuleNotFoundError: No module named 'msgpack'
```

**Pull request merged**